pkg/cache/cache.go

```diff
@@ -166,6 +166,14 @@ type Options struct {
 	//
 	// This is a global setting for all objects, and can be overridden by the ByObject setting.
 	UnsafeDisableDeepCopy *bool
+
+	// FailOnUnknownResource configures the cache to return a ErrResourceNotCached error when a user
+	// requests a resource the cache is not configured to hold. This error is distinct from an
+	// errors.NotFound.
+	//
+	// Defaults to false, which means that the cache will start a new informer
+	// for every new requested resource.
```

Comment:
Might be worth it to note that by default this creates a cluster-wide cache to help folks understand why they might choose to set it to `true` if they don't have the background we do